shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -469,7 +469,19 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrDirectContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  std::unique_ptr<SurfaceFrame> frame) {
-  FML_DCHECK(flutter_view_);
+  // Warning: flutter_view_ could be deleted on the platform thread between load
+  // and retain.  Both are atomic but their composition isn't.
+  fml::scoped_nsobject<UIView> flutter_view([flutter_view_.load() retain]);
+  if (!flutter_view) {
+    // `flutter_view_` is set by the platform thread but read on the raster
+    // thread.  It is possible this is executing on the raster thread before the
+    // flutter_view_ has been set.  In such case we just submit the frame
+    // instead of crashing.  The may cause incorrect frames to render
```

Comment:
This is fine, but I wonder why this is even allowed. The shell setup should ensure that things are wired up on the platform thread before the raster thread starts consuming tasks. Do you know where this is breaking?

Reply:
Yes, please see the attached issue.  TLDR viewDidAppear starts the raster thread, viewDidLayoutSubviews sets up the platforms view controller.  There is no guarantee that viewDidLayoutSubviews is executed before SubmitFrame.  We do need a larger fix.  I outlined our 3 options in the issue.

Reply:
I see. That makes sense. This has nothing to do with the setup in shell after looking at the code.